lib/presentation/shared_widgets/pin_input/pin_input.dart

```diff
@@ -0,0 +1,105 @@
+import 'package:flutter/material.dart';
+
+import 'pin_text_field.dart';
+
+class PinInput extends StatefulWidget {
+  final int pinLength;
+  final bool readOnly;
+  final Function(String) submit;
```

Comment:
I made this required. Makes no sense to have a pin input without a submit callback. It's like having a button without onTap/onPressed

Reply:
Cool, that works too.